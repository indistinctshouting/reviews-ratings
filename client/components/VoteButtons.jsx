import React from 'react';
import Button from './Button.jsx';
import _ from 'lodash';


export default class VoteButtons extends React.Component {
  constructor(props) {
    super(props);
  }
  
  handleButtonClick(count, clicked) {
    if (!count && !clicked) {
      count = 1;
    } else if (!clicked) {
      count += 1;
    } else if (clicked) {
      count -= 1;
    }
    console.log(count, clicked);
    this.setState({
      count,
      clicked: !clicked
    });
  }

  render() {
    return (
      <div className="footer-buttons">
        {
          _.map(this.props.votes, (value, key) => {
            return <Button name={key} count={value} handleButtonClick={this.handleButtonClick}/>;
          })
        }
      </div>
    );
  }
}