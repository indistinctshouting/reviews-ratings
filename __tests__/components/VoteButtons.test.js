import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import VoteButtons from '../../client/components/VoteButtons';


describe('<VoteButtons/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <VoteButtons 
        votes={{upvotes: { useful: 10, funny: 2, cool: 0 }}}
      />
    );
  });

  it('should render Vote Buttons component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render 3 button components', () => {
    expect(wrapper.find('.footer-buttons').children()).toEqual(3);
  });
  
  it('should have a method called handleButtonClick', () => {
    let instance = wrapper.instance();
    expect(typeof instance.handleButtonClick).toBe('function');
  });

});