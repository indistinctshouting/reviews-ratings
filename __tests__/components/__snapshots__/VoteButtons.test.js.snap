// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VoteButtons/> should render Vote Buttons component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VoteButtons
    votes={
      Object {
        "upvotes": Object {
          "cool": 0,
          "funny": 2,
          "useful": 10,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Button
          count={
            Object {
              "cool": 0,
              "funny": 2,
              "useful": 10,
            }
          }
          handleButtonClick={[Function]}
          name="upvotes"
        />,
      ],
      "className": "footer-buttons",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "count": Object {
            "cool": 0,
            "funny": 2,
            "useful": 10,
          },
          "handleButtonClick": [Function],
          "name": "upvotes",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            count={
              Object {
                "cool": 0,
                "funny": 2,
                "useful": 10,
              }
            }
            handleButtonClick={[Function]}
            name="upvotes"
          />,
        ],
        "className": "footer-buttons",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "count": Object {
              "cool": 0,
              "funny": 2,
              "useful": 10,
            },
            "handleButtonClick": [Function],
            "name": "upvotes",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
